pkg/reconciler/openshift/tektonconfig/manifests_post_reconciler.go

```diff
@@ -0,0 +1,219 @@
+	manifestsPostReconcileYamlDirectory = "static/tekton-config/00-postreconcile"
+	// installerSet label value
+	manifestsPostReconcileLabelCreatedByValue = "tekton-config-post-reconcile-manifests"
+	// pipeline console plugin environment variable key
+	PipelineConsolePluginImageEnvironmentKey = "IMAGE_PIPELINE_CONSOLE_PLUGIN"
+	// pipeline console plugin container name, used to replace the image from the environment
+	PipelineConsolePluginContainerName = "pipeline-console-plugin"
+)
+
+var (
+	// label filter to set/get installerSet specific to this reconciler
+	manifestsPostReconcileInstallerSetLabel = metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.InstallerSetType: v1alpha1.ConfigResourceName,
+			v1alpha1.CreatedByKey:     manifestsPostReconcileLabelCreatedByValue,
+		},
+	}
+)
+
+type manifestsPostReconciler struct {
+	logger                     *zap.SugaredLogger
+	operatorClientSet          versioned.Interface
+	syncOnce                   sync.Once
+	resourcesYamlDirectory     string
+	operatorVersion            string
+	pipelineConsolePluginImage string
+	manifest                   mf.Manifest
+}
+
+// reconcile steps
+// 1. get post reconcile (pipeline console plugin) manifests from kodata
+// 2. verify the existing installerSet hash value
+// 3. if there is a mismatch or the installerSet not available, (re)create it
+func (prm *manifestsPostReconciler) reconcile(ctx context.Context, tektonConfigCR *v1alpha1.TektonConfig) error {
+
+	prm.updateOnce(ctx)
+
+	// verify he availability of the installerSet
+	labelSelector, err := common.LabelSelector(manifestsPostReconcileInstallerSetLabel)
+	if err != nil {
+		return err
+	}
+
+	installerSetList, err := prm.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return err
+	}
+
+	doCreateInstallerSet := false
+	var deployedInstallerSet v1alpha1.TektonInstallerSet
+
+	if len(installerSetList.Items) > 1 {
+		for _, installerSet := range installerSetList.Items {
+			err = prm.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().Delete(ctx, installerSet.GetName(), metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+		}
+		doCreateInstallerSet = true
+	} else if len(installerSetList.Items) == 1 {
+		deployedInstallerSet = installerSetList.Items[0]
+	} else {
+		doCreateInstallerSet = true
+	}
+
+	// clone the manifest
+	manifest := prm.manifest.Append()
+	// apply transformations
+	if err := prm.transform(ctx, &manifest, tektonConfigCR); err != nil {
+		tektonConfigCR.Status.MarkNotReady(fmt.Sprintf("transformation failed: %s", err.Error()))
+		return err
+	}
+
+	// get expected hash value of the manifests
+	expectedHash, err := prm.getHash(manifest.Resources())
+	if err != nil {
+		return err
+	}
+
+	if !doCreateInstallerSet {
+		// compute hash from the deployed installerSet
+		deployedHash, err := prm.getHash(deployedInstallerSet.Spec.Manifests)
```

Comment:
I think we should add the label of hash value and rely on that. manifests can be changed for some workaround etc

Reply:
@piyush-garg Thought of taking full control to operator. still do we want to allow to apply workaround with installerset(for console plugin)?